src/Halogen/Menu/Component.purs

```diff
@@ -25,8 +27,8 @@ import Halogen.Menu.Submenu.Component as SUB
 
 data Query a next
   = Set (State a) next
-  | SelectSubmenu Int DET.MouseEvent next
-  | DismissSubmenu DET.MouseEvent next
+  | SelectSubmenu Int (Maybe DET.MouseEvent) next
+  | DismissSubmenu (Maybe DET.MouseEvent) next
   | HandleSubmenu (SUB.Message a) next
```

Comment:
Maybe we could add `HandleMouseEvent DET.MouseEvent (Query a)`, then in the eval it'd be something like:

``` purescript
HandleMouseEvent mbe q ->
  for_ mbe \e → liftEff do
    DEE.preventDefault $ DET.mouseEventToEvent e
    DEE.stopPropagation $ DET.mouseEventToEvent e
  eval q
```

That way the Select/Dismiss submenu stuff can have a more pleasant interface for external triggering.